### PR TITLE
Hidden vs. Readonly Properties

### PR DESCRIPTION
**Pull request by aj-jester**
- Document how `RandomString` behaves based on `hidden` and `readonly` fields.
-  `help_text` docs were missing and clarified usage of `description`.
    - Also added a note that both support Markdown syntax.
- Updated KOTS-K8s compatibility matrix.

### CODE REVIEW

File: content/reference/v1beta1/config.md [hunk 1]
```diff
@@ -235,9 +239,23 @@ Items can be readonly.
           readonly: true
 ```
```
Comment (emosbaugh) [inline content/reference/v1beta1/config.md:241]: My understanding is it’s more than just random string

Comment (aj-jester) [reply content/reference/v1beta1/config.md:241]: @emosbaugh Oh! So does that mean _any_ function that generate "things" (RandomString being just one of them) will regenerate based on `hidden` vs `readonly`?

File: content/reference/template-functions/static-context.md [hunk 2]
```diff
@@ -159,6 +158,13 @@ Or for a total of 64 `a`s and `b`s:
 '{{repl RandomString 64 "[ab]" }}'
 ```
 
+Each time that this function is called, the behavior changes based on the [hidden](/reference/v1beta1/config/#hidden) and [readonly](/reference/v1beta1/config/#readonly) properties.
+
+- To generate `RandomString` value that is **persistent** between Config changes, use it in conjuction with `hidden` property set to `true`. The `value` is not shown in HTML, hence it cannot be modified.
```
Comment (dexhorthy) [inline content/reference/template-functions/static-context.md:163]:
```suggestion
- To generate a `RandomString` value that is **persistent** between Config changes, use it in conjunction with `hidden` property set to `true`. The `value` is not shown in HTML, hence it cannot be modified.
```

File: content/reference/template-functions/static-context.md [hunk 3]
```diff
@@ -159,6 +158,13 @@ Or for a total of 64 `a`s and `b`s:
...
+- To generate `RandomString` value that is **ephemeral** between Config changes, use it in conjuction with `readonly` property set to `true`. The `value` is shown in HTML but the it cannot be modified.
```
Comment (dexhorthy) [inline content/reference/template-functions/static-context.md:164]:
```suggestion
- To generate a `RandomString` value that is **ephemeral** between Config changes, use it in conjunction with `readonly` property set to `true`. The `value` is shown in HTML but the it cannot be modified.
```

File: content/reference/v1beta1/config.md (hunk 1 above)
Comment (dexhorthy) [reply content/reference/v1beta1/config.md:241]: I think that's right @aj-jester but I'll let @emosbaugh confirm

File: content/reference/template-functions/static-context.md [hunk 4]
```diff
@@ -159,6 +158,13 @@ Or for a total of 64 `a`s and `b`s:
 '{{repl RandomString 64 "[ab]" }}'
 ```
 
+Each time that this function is called, the behavior changes based on the [hidden](/reference/v1beta1/config/#hidden) and [readonly](/reference/v1beta1/config/#readonly) properties.
```
Comment (emosbaugh) [inline content/reference/template-functions/static-context.md:161]: does hidden matter? i dont know

File: content/reference/v1beta1/config.md [hunk 5]
```diff
@@ -225,6 +225,10 @@ Items can be hidden. They will not be visible if hidden.
           value: "{{repl RandomString 40}}"
 ```
 
+When used in conjunction with [RandomString](/reference/template-functions/static-context/#randomstring)
```
Comment (emosbaugh) [inline content/reference/v1beta1/config.md:228]: its my experience that readonly affect this. does hidden?

File: content/reference/v1beta1/config.md [hunk 6]
```diff
@@ -235,9 +239,23 @@ Items can be readonly.
           readonly: true
 ```
 
+When used in conjunction with [RandomString](/reference/template-functions/static-context/#randomstring)
```
Comment (emosbaugh) [inline content/reference/v1beta1/config.md:242]: its more than just random string. i think we should generalize this

File: content/reference/template-functions/static-context.md (hunk 4 above)
Comment (aj-jester) [reply content/reference/template-functions/static-context.md:161]: It does not but generally it is used with `hidden`.

File: content/reference/v1beta1/config.md (hunk 6 above)
Comment (aj-jester) [reply content/reference/v1beta1/config.md:242]: Updated to make it more generic.

File: content/reference/v1beta1/config.md (hunk 5 above)
Comment (aj-jester) [reply content/reference/v1beta1/config.md]: It doesn't but it generally is used with `hidden` If its not a hidden field they can edit it.

File: content/reference/v1beta1/config.md (hunk 1 above)
Comment (aj-jester) [reply content/reference/v1beta1/config.md:241]: Updated the doc to be more generic.

File: content/reference/template-functions/static-context.md (hunk 2 above)
Comment (dexhorthy) [reply content/reference/template-functions/static-context.md:163]: @aj-jester I noticed this was resolved but the change was not applied -- can we fix the typos here?

File: content/reference/v1beta1/config.md [hunk 7]
```diff
@@ -235,9 +239,23 @@ Items can be readonly.
...
 ```
+
+### `help_text`
+This is similar to `description` but for `items`. This property can show a helpful message below `title`. [Markdown](https://guides.github.com/features/mastering-markdown/) syntax is supported.
+```yaml
+    - name: toggles
+      items:
+      - name: http_enabled
+        title: HTTP Enabled
+        help_text: When enabled we will listen to http
```
Comment (dexhorthy) [inline content/reference/v1beta1/config.md:259]: kind of unrelated but I'm not totally a fan of this kind of language in a config screen, and I don't think we should encourage vendors to do this in their demo apps. Who is `we` in this case?


how about

```suggestion
        help_text: Check to enable the HTTP listener
```

**Pull request merged**